packages/analytics/amplify_analytics_pinpoint/example/integration_test/utils/mock_secure_storage.dart

```diff
@@ -2,22 +2,35 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import 'dart:async';
+import 'dart:convert';
 
 import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+import 'package:amplify_analytics_pinpoint_example/amplifyconfiguration.dart';
+
 import 'package:amplify_flutter/amplify_flutter.dart';
 
 /// Static key/value storage for use in integration tests.
 ///
-/// Should be static to mimic the behavior of iOS/Android which persist keys
+/// Must be static to mimic the behavior of iOS/Android which persist keys
 /// between relaunches (e.g. between test groups).
-final SecureStorageInterface mockEndpointInfoStore = _MockEndpointInfoStore()
-  ..write(
-    key: EndpointStoreKey.endpointId.name,
-    value: mockEndpointId,
-  );
+final SecureStorageInterface mockPersistedSecuredStorage = () {
+  final json = jsonDecode(amplifyconfig) as Map;
+  final amplifyConfig = AmplifyConfig.fromJson(json.cast());
+  final pinpointAppId =
+      amplifyConfig.analytics!.awsPlugin!.pinpointAnalytics.appId;
+
+  final storage = _MockSecureStorage()
+    ..write(
+      key: pinpointAppId + EndpointStoreKey.endpointId.name,
```

Comment:
This logic should be consolidated somewhere - it's easy to mess up and could potentially change. Check out how it's handled in Auth: https://github.com/aws-amplify/amplify-flutter/blob/next/packages/auth/amplify_auth_cognito_dart/lib/src/credentials/cognito_keys.dart

Reply:
Or expose `_EndpointStore` - just saw that

Reply:
Switched to a new MockEndpointStore class.

Reply:
Switched to a new MockEndpointStore class.